pkg/ansible/runner/runner.go

```diff
@@ -68,14 +68,18 @@ type cmdFuncType func(ident, inputDirPath string, maxArtifacts, verbosity int) *
 
 func playbookCmdFunc(path string) cmdFuncType {
 	return func(ident, inputDirPath string, maxArtifacts, verbosity int) *exec.Cmd {
-		return exec.Command("ansible-runner", ansibleVerbosityString(verbosity), "--rotate-artifacts", fmt.Sprintf("%v", maxArtifacts), "-p", path, "-i", ident, "run", inputDirPath)
+		ansibleVerbosity := ansibleVerbosityString(verbosity)
+		rotateArtifacts :=  fmt.Sprintf("%v", maxArtifacts)
+		return exec.Command("ansible-runner", ansibleVerbosity, "--rotate-artifacts",rotateArtifacts, "-p", path, "-i", ident, "run", inputDirPath)
 	}
 }
 
 func roleCmdFunc(path string) cmdFuncType {
 	rolePath, roleName := filepath.Split(path)
 	return func(ident, inputDirPath string, maxArtifacts, verbosity int) *exec.Cmd {
-		return exec.Command("ansible-runner", ansibleVerbosityString(verbosity), "--rotate-artifacts", fmt.Sprintf("%v", maxArtifacts), "--role", roleName, "--roles-path", rolePath, "--hosts", "localhost", "-i", ident, "run", inputDirPath)
+		ansibleVerbosity := ansibleVerbosityString(verbosity)
+		rotateArtifacts :=  fmt.Sprintf("%v", maxArtifacts)
+		return exec.Command("ansible-runner", ansibleVerbosity, "--rotate-artifacts", rotateArtifacts, "--role", roleName, "--roles-path", rolePath, "--hosts", "localhost", "-i", ident, "run", inputDirPath)
```

Comment:
-1 I would not fix this

Reply:
This is about security. See: 

```
internal/scorecard/interfaces.go:51: G204: Subprocess launched with function call as argument or cmd arguments (gosec)
	cmd := exec.Command(p.config.Command, p.config.Args...)
pkg/ansible/controller/reconcile.go:75: G404: Use of weak random number generator (math/rand instead of crypto/rand) (gosec)
	ident := strconv.Itoa(rand.Int())
make: *** [test-linter] Error 1
```
So, in your opinion would not use this lint check or you just not think that we should not do this change?

Reply:
So gosec G204  https://github.com/securego/gosec/blob/53be8dd8644ee48802114178cff6eb7e29757414/issue.go#L59 maps to CWE 78:
https://cwe.mitre.org/data/definitions/78.html

After reading CWE 78, the intent is to protect commands from unintended consequences and attacks. `gosec` flagged this because we are using functions for arguments. These functions are well protected in my opinion. One is a `fmt.Sprintf` that's an internal go api, if that is being attacked we've got bigger issues. The second function is `ansibleVerbosityString` that one is actually taking arguments from the user and converting them. 

So I think we should flag these as don't fix. I would make gosec optional for builds. I would rather run gosec manually occasionally and review the output. Because the change you made would not have fixed this if those functions were really attackable.